node/libs/storage/src/block_store.rs

```diff
@@ -0,0 +1,218 @@
+//! Defines storage layer for finalized blocks.
+use anyhow::Context as _;
+use std::{collections::BTreeMap, fmt, sync::Arc};
+use zksync_concurrency::{ctx, sync};
+use zksync_consensus_roles::validator;
+
+/// State of the `BlockStore`: continuous range of blocks.
+#[derive(Debug, Clone)]
+pub struct BlockStoreState {
+    /// Stored block with the lowest number.
+    pub first: validator::CommitQC,
+    /// Stored block with the highest number.
+    pub last: validator::CommitQC,
```

Comment:
Bikeshedding: There's some inconsistency regarding "first / last" vs "lowest / highest" block numbers. Personally, I like "first / last" terms slightly better.

Reply:
wdym by inconsistency? first block in storage is the one with the lowest block number by definition (last analogically)

Reply:
Space of block numbers is fixed - there are block numbers without a block, so it is hard to say that last block number is the same as the number of the last block. And you could hardly call it a definition then.